packages/core/src/components/WarningPanel/WarningPanel.tsx

```diff
@@ -29,57 +37,103 @@ const ErrorOutlineStyled = () => {
   const classes = useErrorOutlineStyles();
   return <ErrorOutline classes={classes} />;
 };
+const ExpandMoreIconStyled = () => {
+  const classes = useErrorOutlineStyles();
+  return <ExpandMoreIcon classes={classes} />;
+};
 
 const useStyles = makeStyles<BackstageTheme>(theme => ({
-  message: {
-    display: 'flex',
-    flexDirection: 'column',
-    padding: theme.spacing(1.5),
+  panel: {
+    // display: 'flex',
+    // flexDirection: 'column',
+    // padding: theme.spacing(1.5),
     backgroundColor: theme.palette.warningBackground,
     color: theme.palette.warningText,
     verticalAlign: 'middle',
   },
-  header: {
+  summary: {
     display: 'flex',
     flexDirection: 'row',
-    marginBottom: theme.spacing(1),
   },
-  headerText: {
+  summaryText: {
     color: theme.palette.warningText,
+    fontWeight: 'bold',
   },
-  messageText: {
+  message: {
+    width: '100%',
+    display: 'block',
     color: theme.palette.warningText,
+    backgroundColor: theme.palette.warningBackground,
+  },
+  details: {
+    width: '100%',
+    display: 'block',
+    color: theme.palette.textContrast,
+    backgroundColor: theme.palette.background.default,
+    border: `1px solid ${theme.palette.border}`,
+    padding: theme.spacing(2.0),
+    fontFamily: 'sans-serif',
   },
 }));
 
-/**
- * WarningPanel. Show a user friendly error message to a user similar to ErrorPanel except that the warning panel
- * only shows the warning message to the user
- */
-
 type Props = {
-  message?: React.ReactNode;
   title?: string;
+  severity?: 'warning' | 'error' | 'info';
+  message?: React.ReactNode;
   children?: React.ReactNode;
 };
 
+const capitalize = (s: string) => {
+  return s.charAt(0).toUpperCase() + s.slice(1);
+};
+
+/**
+ * WarningPanel. Show a user friendly error message to a user similar to ErrorPanel except that the warning panel
+ * only shows the warning message to the user.
+ *
+ * @param {string} [severity=warning] Ability to change the severity of the alert. Not fully implemented. (error, warning, info)
+ * @param {string} [title] A title for the warning. If not supplied, "Warning" will be used.
```

Comment:
does this mean that it would show just "Warning" with no further info, and make me expand it to see the actual details? 🤔  Maybe in that case it's better to show the message only, right? And maybe even not as an accordion, because it would contain all the text already.

I guess I'm unclear on the difference between message and children. Today it seems like they are equals?

Reply:
Yeah. So the current component already  had the three, and basically it was "title" at the top, and message + children would get squeezed together inside the component. (in my view, potentially reducing the value of a separation)

Throughout the current code base, various permutations are used, so I tried to keep it flexible.

At it's most simple...

![image](https://user-images.githubusercontent.com/33203301/106296215-600e7300-621f-11eb-9360-a501ebe601f1.png)

Each being optional. So forget the message, then it's just a title and a white box. Add children, it gets put into the white box. Forget a title, it defaults to just "Warning" (and we do have some usages of it without a title in the code base).

I opted to always prefix Warning to it so it sort of enforces a commonality of it being warning, but could potentially remove and ONLY use as a default fallback if we want?